src/QsCompiler/CompilationManager/EditorSupport/SymbolInfo.cs

```diff
@@ -277,13 +342,168 @@ bool OverlapsWithPosition(Range symRange)
             {
                 // the given position corresponds to a variable declared as part of a specialization declaration or implementation
                 var defStart = defOffset + defRange.Start;
-                var statements = implementation.StatementsAfterDeclaration(defStart - specPos);
+                var statements = SplitStatementsByPosition(implementation, defStart - specPos).Item3;
                 var scope = new QsScope(statements.ToImmutableArray(), locals);
                 referenceLocations = IdentifierReferences.Find(definition.Item.Item1, scope, file.FileName, specPos).Select(AsLocation);
             }
 
             declarationLocation = AsLocation(file.FileName, definition.Item.Item2, defRange);
             return true;
         }
+
+        /// <summary>
+        /// Finds the innermost scope that contains <paramref name="position"/> and returns the scope, statements in the
+        /// innermost scope before and including <paramref name="position"/>, and statements after
+        /// <paramref name="position"/>. If <paramref name="position"/> points to a statement that declares variables
+        /// and contains a child scope, the statements after <paramref name="position"/> are from the child scope.
+        /// </summary>
+        /// <param name="scope">A scope that contains <paramref name="position"/>.</param>
+        /// <param name="position">The position to split by.</param>
+        /// <returns>The innermost scope and statements before and after <paramref name="position"/>.</returns>
+        private static (QsScope, ImmutableList<QsStatement>, ImmutableList<QsStatement>) SplitStatementsByPosition(
```

Comment:
How does it deal with statements whose position is Null? That would be good to document.

Reply:
I think it has the same behavior as the original `StatementsAfterAndLocalDeclarationsAt` that it's replacing: it ignores any statement without a location, *and* any statements after it (because of the `TakeWhile`). So it basically treats statements without a location as if it occurred after the given position.

Added comment.